2024-04-23 08:07:28,901: asyncio: INFO: [MLTrader] Executing the initialize lifecycle method
2024-04-23 08:07:30,551: root: INFO: Current backtesting datetime 2023-01-03 08:30:00-05:00
2024-04-23 08:07:30,552: asyncio: INFO: [MLTrader] Executing the before_market_opens lifecycle method
2024-04-23 08:07:30,553: root: INFO: Current backtesting datetime 2023-01-03 09:30:00-05:00
2024-04-23 08:07:30,553: asyncio: INFO: [MLTrader] Executing the before_starting_trading lifecycle method
2024-04-23 08:07:30,554: asyncio: INFO: [MLTrader] [34mExecuting the on_trading_iteration lifecycle method at 2024-04-23 08:07:30[0m
2024-04-23 08:07:30,554: asyncio: ERROR: [MLTrader] 'list' object is not callable
2024-04-23 08:07:30,573: asyncio: ERROR: [MLTrader] Traceback (most recent call last):
  File "C:\Users\Osama Warsi\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 919, in run
    self._run_trading_session()
  File "C:\Users\Osama Warsi\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 867, in _run_trading_session
    self._on_trading_iteration()
  File "C:\Users\Osama Warsi\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 262, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Osama Warsi\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 286, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Osama Warsi\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 397, in _on_trading_iteration
    raise e
  File "C:\Users\Osama Warsi\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 369, in _on_trading_iteration
    on_trading_iteration()
  File "C:\Users\Osama Warsi\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\tools\decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Osama Warsi\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\tools\decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "D:\VVWoodworks Market Trader\AI-Bot-Trader\MarketBot.py", line 141, in on_trading_iteration
    for items in self.symbol():
                 ^^^^^^^^^^^^^
TypeError: 'list' object is not callable

2024-04-23 08:07:30,602: asyncio: INFO: [MLTrader] Executing the on_bot_crash event method
2024-04-23 08:07:30,624: root: INFO: Backtesting finished
2024-04-23 08:07:31,933: asyncio: WARNING: [MLTrader] Cannot plot returns because the strategy returns are missing
2024-04-23 08:07:32,057: asyncio: WARNING: [MLTrader] Cannot create a tearsheet because the strategy returns are missing
2024-04-23 08:07:32,057: backtest_stats: INFO: Backtest took 0:00:03.158969 for a speed of 0.000
